app/channel.go

```diff
@@ -354,6 +354,24 @@ func (a *App) UpdateChannel(channel *model.Channel) (*model.Channel, *model.AppE
 	}
 }
 
+func (a *App) UpdateChannelScheme(channel *model.Channel) (*model.Channel, *model.AppError) {
+	var oldChannel *model.Channel
+	var err *model.AppError
+	if oldChannel, err = a.GetChannel(channel.Id); err != nil {
+		return nil, err
+	}
+
+	a.InvalidateCacheForChannel(channel)
+
+	oldChannel.SchemeId = channel.SchemeId
+
+	if result := <-a.Srv.Store.Channel().Update(oldChannel); result.Err != nil {
```

Comment:
Probably here makes sense to use the a.UpdateChannel directly, do exactly the same in the database and also send the websocket update of the channel (something needed for the correct behavior in the frontend).

Reply:
@jespino Good idea; cleaner. Switched.